daemon/cluster/cluster.go

```diff
@@ -446,11 +446,20 @@ func (c *Cluster) UnlockSwarm(req types.UnlockRequest) error {
 
 	c.mu.RLock()
 	state := c.currentNodeState()
+	if !state.IsActiveManager() {
+		if err := c.errNoManager(state); err != errSwarmLocked {
+			c.mu.RUnlock()
+			return err
+		}
+	}
+
 	nr := c.nr
 	c.mu.RUnlock()
+
 	if nr == nil || errors.Cause(state.err) != errSwarmLocked {
```

Comment:
When active manager, `nr == nil` does not meet, nor `errors.Cause(state.err) != errSwarmLocked`. 

In a word, this way seems confusing, I will change to another way to achieve this.